### DISCUSSION TITLE
improving the quickstart tutorial ?

### DISCUSSION THREAD

**Discussion by moolight-seashell** [slint-ui/slint#5785] [General]
before the subject i want to say that i love slint a lot , 

the thing actually i find that the docs quickstart tutorial is not is not very well designed.
(anywayin thanks for the slint pad and the very great documentation of the slint laguange)

on the two first step we make our first tile , great.
then, just after, on the third step there is a big step in complexity.
we have custom properties and the tuto never talk about these before.
we have animations even if we're only at the third step of a complete beginer tutorial.
(animation is typically the kind of subject I would see at the end of the tutorial as an optional step.)

I've been thinking about doing another tutorial for a while now, 
(only if I find the time to do it =)
 and I'd like to know what you think about this ?

**Comment by tronical** [comment]
Thank you for the kind words and constructive criticism. For another tutorial, what kind of target UI do you have in mind? Or would it perhaps make sense to introduce additional steps in the existing one instead?

**Reply by moolight-seashell** [reply]
Hey thanks you for the reply

I think adding a step between current 2 and 3 to preset the custom property would be great.

personally I would have seen application with multiple screen
and i think this can be helpful to  introduce properties and callbacks fairly quickly.
(typically you have a blue screen and a red screen and a button to switch between them )

this kind of code is not only quite simple, 
but looks a lot like an "application controller" typical of mvc2 apps.

and also because it's the kind of code you'd expect to find a lot in projects like
from energy monitor :
```slint
    if root.screen-size == ScreenSize.Mobile : MobileMain {
        preferred-width: 100%;
        preferred-height: 100%;
    }

    if root.screen-size == ScreenSize.EmbeddedMedium : MidMain {
        preferred-width: 100%;
        preferred-height: 100%;
    }
```

personally, I was thinking of something like an application with two screen. fist a todo-list screen and then a notes screen. the thing is i don't like this idea, it's a bit boring and maybe to complicated for a Quickstart.

but yep, maybe adding a step to the tutorial would be enough.

**Reply by moolight-seashell** [reply]
and maybe add an adroid (+ embeded ?) quickstart ?

**Reply by tronical** [reply]
Thanks for the feedback. I agree, generally.

I wonder what application that could be that has multiple screens like that? Maybe turn the [todo-mvc](https://snapshots.slint.dev/master/demos/todo-mvc/) into a tutorial?

**Reply by moolight-seashell** [reply]
definitely , this looks nice
